CMakeLists.txt

```
@@ -172,7 +182,7 @@ if(NOT MSVC)
   add_c_compiler_flag_if_supported(OUR_FLAGS_OWN -Wduplicated-branches)
   add_c_compiler_flag_if_supported(OUR_FLAGS_OWN -Wlogical-op)
   add_c_compiler_flag_if_supported(OUR_FLAGS_OWN -Wrestrict)
-  add_c_compiler_flag_if_supported(OUR_FLAGS_OWN -Wuseless-cast)
+  add_cxx_compiler_flag_if_supported(CMAKE_CXX_FLAGS "-Wuseless-cast")
```

To only add this flag to our own compilation units, do
```suggestion
  if(CMAKE_VERSION VERSION_GREATER 3.3 OR CMAKE_VERSION VERSION_EQUAL 3.3)
    add_c_compiler_flag_if_supported(OUR_FLAGS_OWN
      $<$<COMPILE_LANGUAGE:CXX>:-Wuseless-cast>
      -Wuseless-cast
    )
  endif()
```